CONTRIBUTING.md

```diff
@@ -0,0 +1,65 @@
+## Contributing guide
+A simple four step guide to consider when you want your pull request to be merged without a hassle. This guide mainly focusses on the proper use of Git. It has some overlap with the more general information found in the [Development Policy File](https://github.com/arduino/Arduino/wiki/Development-Policy).
+
+### 1. Bugs and small enhancements
+To report a bug or a small enhancement please use the [issue tracker](http://github.com/arduino/Arduino/issues).
+
+### 2. Larger changes
+Before starting to work on bigger topics like modifying the API or changes with backward compatibility trade-offs please discuss them in the [mailing list](https://groups.google.com/a/arduino.cc/forum/#!forum/developers) first.  
+
+### 3. Commit messages
+An easy to read pull request will speed up the merging process. Your commit messages need to be logically separate. And containing enough information on their own. When this is done consistently your pull request will have an easy to read log of changes.
+
+If you did not read the following 7 points before or just want to fresh up. Please read up on them on this [website](https://chris.beams.io/posts/git-commit) by Chris Beams.
+
+1. Separate subject from body with a blank line
+2. Limit the subject line (first line) to 50 characters
+3. Capitalize the subject line
+4. Do not end the subject line with a period `(.)`
+5. Use the imperative mood in the subject line.
+This should be in the written as giving an instruction for example "update getting started documentation" (it shows what the PR achieves when merging it)
```

Comment:
"update" should be capitalized (see point 3). I actually don't think this is the ideal example of a subject because the term "update" is much too general.